### PR TITLE
Add tests for marathon.go and refactor

### PR DESCRIPTION
**Pull request by janisz**

### CODE REVIEW

File: apps/app_test.go [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+				"taskId": "myapp.c8b449f0-9812-11e5-a06e-56847afe9799"
+			}]
+		}],
+		"lastTaskFailure": null
+	}
+}`)
+
+	app, err := ParseApp(appBlob)
+	assert.Nil(t, err)
+	assert.Equal(t, len(app.Tasks), 2)
```
Comment (dankraw) [inline apps/app_test.go]: `assert.Len()` ;)

Comment (janisz) [inline apps/app_test.go]: Fixed

File: apps/app_test.go [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+				"version": "2015-06-24T14:56:57.466Z"
+			}
+		]
+    }
+]}
+`)
+
+	apps, err := ParseApps(appBlob)
+	assert.Nil(t, err)
+	assert.Equal(t, len(apps), 1)
```
Comment (dankraw) [inline apps/app_test.go]: maybe some app[field] check would be nice here or we should limit this json-string to necessary minimum (or maybe reading it from external file)

File: apps/app_test.go [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+				"consecutiveFailures": 0,
+				"firstSuccess": "2015-12-01T10:03:37.313Z",
+				"lastFailure": null,
+				"lastSuccess": "2015-12-01T10:03:42.337Z",
+				"taskId": "myapp.c8b449f0-9812-11e5-a06e-56847afe9799"
+			}]
+		}],
+		"lastTaskFailure": null
+	}
+}`)
```
Comment (dankraw) [inline apps/app_test.go]: same here, we could try to read it from external file

File: marathon/marathon_test.go [hunk 4]
```diff
@@ -1,246 +1,303 @@
...
-			"minimumHealthCapacity": 1.0,
-			"maximumOverCapacity": 1.0
-		},
-		"labels": {
-			"consul": "true",
-			"public": "tag"
+func TestMarathon_AppsWhenMarathonReturnEmptyList(t *testing.T) {
+	t.Parallel()
+	// given
+	server, transport := stubServer("/v2/apps?embed=apps.tasks", `{"apps": []}`)
```
Comment (dankraw) [inline marathon/marathon_test.go]: :cool:

**Comment by dankraw** [comment]
great :+1:

File: apps/app_test.go (hunk 3 above)
Comment (janisz) [inline apps/app_test.go]: Done

File: apps/app_test.go (hunk 2 above)
Comment (janisz) [inline apps/app_test.go]: I deleted unused fields in structs and move JSON to file

**Pull request merged**